cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/AppFabricServiceMain.java

```diff
@@ -158,6 +159,10 @@ protected void addServices(Injector injector, List<? super Service> services,
     services.add(new RetryOnStartFailureService(() -> injector.getInstance(DatasetService.class),
                                                 RetryStrategies.exponentialDelay(200, 5000, TimeUnit.MILLISECONDS)));
     services.add(injector.getInstance(AppFabricServer.class));
+    services.add(new RetryOnStartFailureService(
+      () -> new NamespaceInitializerService(injector.getInstance(NamespaceQueryAdmin.class),
```

Comment:
create the initializer with the injector instead of manually

Reply:
fixed